src/components/ConversationView/index.tsx

```diff
@@ -178,15 +178,17 @@ const ConversationView = () => {
         const tables = await connectionStore.getOrFetchDatabaseSchema(
           connectionStore.currentConnectionCtx?.database
         );
-        // Empty table name(such as "") denote all table. "" and `undefined` both are false in `if`
-        if (currentConversation.tableName) {
-          const table = tables.find((table) => {
-            return table.name === currentConversation.tableName;
+        // Empty table name(such as []) denote all table. [] and `undefined` both are false in `if`
+        if (currentConversation.selectedTablesName) {
+          currentConversation.selectedTablesName.forEach((tableName) => {
+            const table = tables.find((table) => table.name === tableName);
+            if (table) {
+              if (tokens < MAX_TOKENS / 2) {
+                tokens += countTextTokens(schema + table.structure);
+                schema += table.structure;
```

Comment:
Let's refactor it a bit.

1. first compile the table list based on whether there is selectedTablesName

2. count the token and append the table structure to the schema

Reply:
what does "compile the table list" mean?👀

Reply:
Build the table list